webapp/channels/src/components/start_trial_form_modal/index.tsx

```diff
@@ -0,0 +1,315 @@
+
+import {makeAsyncComponent} from 'components/async_load';
+import useGetTotalUsersNoBots from 'components/common/hooks/useGetTotalUsersNoBots';
+import {COUNTRIES} from 'utils/countries';
+
+import {ModalIdentifiers, TELEMETRY_CATEGORIES} from 'utils/constants';
+
+import './start_trial_form_modal.scss';
+import Input, {SIZE, CustomMessageInputType} from 'components/widgets/inputs/input/input';
+import DropdownInput from 'components/dropdown_input';
+import {RequestLicenseBody} from '@mattermost/types/config';
+import StartTrialFormModalResult from './failure_modal';
+import ExternalLink from 'components/external_link';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/common';
+import {trackEvent} from 'actions/telemetry_actions';
+
+// TODO: Handle embargoed entities explicitly https://mattermost.atlassian.net/browse/MM-51470
+
+const TrialBenefitsModal = makeAsyncComponent('TrialBenefitsModal', React.lazy(() => import('components/trial_benefits_modal/trial_benefits_modal')));
+
+enum TrialLoadStatus {
+    NotStarted = 'NOT_STARTED',
+    Started = 'STARTED',
+    Success = 'SUCCESS',
+    Failed = 'FAILED'
+}
+
+export enum OrgSize {
+    ONE_TO_50 = '1-50',
+    FIFTY_TO_100 = '51-100',
+    ONE_HUNDRED_TO_500 = '101-500',
+}
+
+type Props = {
+    onClose?: () => void;
+    page?: string;
+}
+
+function StartTrialFormModal(props: Props): JSX.Element | null {
+    const [status, setLoadStatus] = useState(TrialLoadStatus.NotStarted);
+    const dispatch = useDispatch<DispatchFunc>();
+    const currentUser = useSelector(getCurrentUser);
+    const [name, setName] = useState('');
+    const [email, setEmail] = useState(currentUser.email);
+    const [companyName, setCompanyName] = useState('');
+    const [orgSize, setOrgSize] = useState<OrgSize | undefined>();
+    const [country, setCountry] = useState('');
+    const [businessEmailError, setBusinessEmailError] = useState<CustomMessageInputType | undefined>(undefined);
+    const {formatMessage} = useIntl();
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.START_TRIAL_FORM_MODAL));
+    const totalUsers = useGetTotalUsersNoBots(true) || 0;
+
+    const handleValidateBusinessEmail = async (email: string) => {
+        if (!email) {
+            setBusinessEmailError(undefined);
+            return;
+        }
+        const isBusinessEmail = await validateBusinessEmail(email)();
+
+        if (isBusinessEmail) {
+            setBusinessEmailError(undefined);
+            return;
+        }
+
+        setBusinessEmailError({
+            type: 'error',
+            value: formatMessage({id: 'start_trial_form.invalid_business_email', defaultMessage: 'Please enter a valid business email address.'},
+            ),
+        } as CustomMessageInputType);
+    };
+
+    useEffect(() => {
+        trackEvent(TELEMETRY_CATEGORIES.SELF_HOSTED_START_TRIAL_MODAL, 'form_opened');
+        if (email) {
```

Comment:
This works since we only want run once behavior, but breaks the "rules of hooks" since it doesn't include the email in the depencency array. To be totally conformant with that, I'd recommend adding a "didOnce" boolean state flag

```ts
const [email, setEmail] = useState(currentUser.Email);
const [didOnce, setDidOnce] = useState(false);

    useEffect(() => {
        if (didOnce) {
            return;
        }
        setDidOnce(true);
        trackEvent(TELEMETRY_CATEGORIES.SELF_HOSTED_START_TRIAL_MODAL, 'form_opened');
        if (email) {
                    handleValidateBusinessEmail(email);
        }
    }, [email, doOnce]);
```